polars/polars-lazy/polars-plan/src/dsl/functions.rs

```diff
@@ -898,52 +898,39 @@ pub fn sum_exprs<E: AsRef<[Expr]>>(exprs: E) -> Expr {
 
 /// Get the the maximum value per row
 pub fn max_exprs<E: AsRef<[Expr]>>(exprs: E) -> Expr {
-    let exprs = exprs.as_ref().to_vec();
-    max_exprs_impl(exprs)
-}
-
-fn max_exprs_impl(mut exprs: Vec<Expr>) -> Expr {
-    if exprs.len() == 1 {
-        return std::mem::take(&mut exprs[0]);
+    let mut exprs = exprs.as_ref().to_vec();
+    if exprs.is_empty() {
+        return Expr::Columns(Vec::new());
     }
 
-    let first = std::mem::take(&mut exprs[0]);
-    first
-        .map_many(
-            |s| {
-                let s = s.to_vec();
-                let df = DataFrame::new_no_checks(s);
-                df.hmax().map(|s| s.unwrap())
-            },
-            &exprs[1..],
-            GetOutput::super_type(),
-        )
-        .alias("max")
+    let func = |s1, s2| {
+        let df = DataFrame::new_no_checks(vec![s1, s2]);
+        df.hmax().map(|s| s.unwrap())
+    };
+    let init = match exprs.pop() {
+        Some(e) => e,
+        // use u32 as that is not cast to float as eagerly
+        _ => lit(Series::new_empty("", &DataType::Int32)),
+    };
+    fold_exprs(init, func, exprs).alias("max")
```

Comment:
I think we should use `reduce_expr` here. That one always takes the first column and saves a redundant operation.

And it is also simpler because we don't need to come up with a `init` value.